ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -113,8 +114,18 @@ private[apiserver] final class ApiSubmissionService private[services] (
       logger.trace(s"Commands: ${request.commands.commands.commands}")
       ledgerConfigurationSubscription
         .latestConfiguration()
-        .map(deduplicateAndRecordOnLedger(seedService.nextSeed(), request.commands, _))
-        .getOrElse(Future.failed(ErrorFactories.missingLedgerConfig()))
+        .fold(Future.failed[Unit](ErrorFactories.missingLedgerConfig()))(ledgerConfiguration => {
```

Comment:
Honestly I think `match` would be easier to read here.

Reply:
I'm a lot more used to `fold` and find it easier to read as it flows linearly, but aggree it's not always the case so I'll change to a match.

Reply:
Change to match, I find the code a lot less readable